client/src/pages/Donations.jsx

```diff
@@ -1,36 +1,180 @@
+  const [errors, setErrors] = useState({});
+  const [contributions, setContributions] = useState([]);
 
-    const addContributions = () => {
-        
-    }
+  const refreshDonations = useCallback(async () => {
+    const res = await axios.get("/api/donations", {
+      params: {
+        userId: user.uid,
+      },
+    });
+    setContributions(res.data);
+  }, [user.uid]);
+
+  //get all donations that the user currently has
+  useEffect(() => {
+    refreshDonations();
+  }, [user, refreshDonations]);
+
+  // finds form errors
+  const findFormErrors = () => {
+    const { name, amount } = donationData;
+    const newErrors = {};
+    // name errors
+    if (name === "") newErrors.name = "cannot be blank!";
+    // donation errors
+    if (amount === 0 || amount === NaN || amount === "")
+      newErrors.amount = "cannot be blank!";
 
-    const generateContributions = () => {
-        let temp = [];
-        for (const [key, value] of Object.entries(contributions)) {
-            temp.push(<tr><td>{key}</td><td>{value}</td></tr>);
-            console.log(key, value);
-        }
-        return temp;
+    return newErrors;
+  };
+
+  const handleChange = (e) => {
+    setDonationData({
+      ...donationData,
+      [e.target.name]: e.target.value,
+    });
+    // Check and see if errors exist, and remove them from the error object:
+    if (!!errors[e.target.name])
+      setErrors({
+        ...errors,
+        [e.target.name]: null,
+      });
+  };
+
+  //should make an axios call to add a donation to the user
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    const newErrors = findFormErrors();
+    if (Object.keys(newErrors).length > 0) {
+      // We got errors!
+      setErrors(newErrors);
+    } else {
+      // No errors! Put any logic here for the form submission!
+      await axios.post("/api/donations", null, {
+        params: {
+          orgName: donationData.name,
+          donationAmount: donationData.amount,
+          userId: user.uid,
+        },
+      });
+      refreshDonations();
     }
+  };
+
+  //pulls id from svg tag and then refreshes donations
+  const handleDelete = async (e) => {
+    await axios.delete("/api/donations/" + e.target.id, {
+      params: {
+        donationId: e.target.id,
+      },
+    });
+    refreshDonations();
+  };
 
-    return (
-        <Container>
-            <h2 className="mx-auto" style={{width:"50%", paddingTop: 60, marginTop: "5%",}}>My Donations:</h2>
-            Total Contributions: 
-            <Table striped bordered hover style={{width: "31.8vw", minWidth: "250px" , margin: "auto"}}>
-            <tbody>
-            {generateContributions()}
-            </tbody>
-            </Table>                   
+  // parses through contributions and sums up results
+  const getTotalContributions = () => {
+    let retVal = 0;
+    contributions.forEach((contribution) => {
+      retVal += parseInt(contribution.donationAmount);
+    });
+    return retVal;
```

Comment:
As an alternative, you can also use the [`reduce`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce) function here